### PR TITLE
Added posts and comments feed to the profile page

### LINKED ISSUE #1126: Profile: posts and comments feed
As a user, I'd like to be able to easily see the history of posts and comments made by another user. 

We will display a feed of posts and comments on users' profile pages. 

See https://invis.io/9EN908ZQPAX#/316071207_Profile_With_Comments

### questions
- Can we do infinite scroll, similar to the channel listing?
- Is there a PRAW API, or should we use elasticsearch?

### acceptance criteria
- [ ] API for fetching list of user's posts, comments
- [ ] tabbed UI
- [ ] count of posts and comments

### PR DESCRIPTION
**Pull request by gsidebo**
#### Pre-Flight checklist

- [X] Screenshots and design review for any changes that affect layout or styling
  - [X] Desktop screenshots 
  - [x] Mobile width screenshots 
  - [ ] tag @ferdi or @pdpinch for review 
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1126

#### What's this PR do?
Adds the posts and comments feed to a user's profile page

#### How should this be manually tested?
Go to a profile page, scroll down to the posts/comments feed, click between the tabs, scroll to the bottom of the page to trigger infinite scrolling, and click around the actual 

#### Any background context you want to provide?
- The comments feed is not very performant at the moment. A new issue will be created to address this.
- There is some code repetition here with the search results page. A new issue will be created to address that too.
- I also tweaked the `PostProxy` query to prefetch `Article` objects. There was an n+1 query happening with those on channel pages and the home page

#### Screenshots (if appropriate)
ss 2018-12-21 at 12 15 52
ss 2018-12-21 at 12 15 46

Mobile:
ss 2019-01-14 at 12 38 20
ss 2019-01-14 at 12 38 12
ss 2019-01-14 at 12 38 05

### CODE REVIEW

**Comment by aliceriot** [comment]
the only thing I'm wondering about the UI is whether the comments should live at a different url - not necessarily because the user would want to link to someone's comments feed (although maybe, idk) but more because clicking the 'comments' link, and switching over to the comments feed, feels like an interaction I should be able to use the back button to undo. Right now the back button will, after clicking over the comments feed, send you back to whatever page you were on before you navigated to the profile.

Anyhow, I'm not sure what we want to do, I was just thinking it felt a bit jarring to have the back button pull me out of the profile page, instead of taking me back to the profile feed as I expected.

@pdpinch

**Comment by aliceriot** [comment]
ui looks great other than that question though! all the buttons and so on that I've tried work and everything.

**Comment by gsidebo** [comment]
Chatted with Alice about the options for putting each tab at a separate URL. Looking into nested routes now

**Comment by pdpinch** [comment]
While it would be nice to have each tab at a separate URL, if it’s a big lift we could leave it for a follow-up PR. 

My rule of thumb — if you can do it today, do it. If you can’t do it today, make a new issue and merge this as is.

**Comment by aliceriot** [comment]
functionality looks great! reading code now

**Comment by gsidebo** [comment]
@aliceriot Ready for review again

File: static/js/containers/ProfileContributionFeed.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const { contributions, selectedTab } = this.props
+
+    if (
+      selectedTab !== prevProps.selectedTab &&
+      contributions &&
+      contributions[selectedTab].loaded === false
```
[the remark this answers is not in the document]
Comment (gsidebo) [reply static/js/containers/ProfileContributionFeed.js]: Good catch. I meant to handle that case and it slipped my mind. I'm just using the endpoint `processing` property.

**Pull request merged**